src/test/java/com/metamx/common/parsers/JavascriptParserTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2011,2012 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.common.parsers;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import junit.framework.Assert;
+import org.junit.Test;
+
+import java.util.Map;
+
+/**
+ */
+public class JavascriptParserTest
+{
+  @Test
+  public void testParse()
+  {
+    final String function = "function(str) { return str.split(\"-\"); }";
+
+    final Parser<String, Object> parser = new JavascriptParser(
+        function,
+        Optional.of("@")
+    );
+    String data = "foo-val1@val2";
+
+    final Map<String, Object> parsed = parser.parse(data);
+    ImmutableMap.Builder builder = ImmutableMap.builder();
+    builder.put("column_1", "foo");
+    builder.put("column_2", Lists.newArrayList("val1", "val2"));
+    Assert.assertEquals(
+        "result",
+        builder.build(),
+        parsed
+    );
+  }
+
+  @Test
+  public void testParse2()
+  {
+    final String function = "function(str) { return \"foo\"; }";
+
+    final Parser<String, Object> parser = new JavascriptParser(
+        function,
+        Optional.<String>absent()
+    );
+    String data = "zoo-val1@val2";
+
+    final Map<String, Object> parsed = parser.parse(data);
+    ImmutableMap.Builder builder = ImmutableMap.builder();
+    builder.put("column_1", "foo");
+    Assert.assertEquals(
+        "result",
+        builder.build(),
+        parsed
+    );
+  }
+}
```

Comment:
can you either add another UT or update this so that field names are explicitly specified and resulting map has something else and not "column_1"?